lib/engine/game/g_18_india/step/track.rb

```diff
@@ -1,15 +1,110 @@
+          include GaugeChangeBorder
+
+          # modified to add remove gauge change option in phase IV
+          def actions(entity)
+            return [] unless entity == current_entity
+            return [] unless @round.removed_gauge.empty?
+            return [] if entity.company? || !can_lay_tile?(entity)
+
+            actions = %w[lay_tile pass]
+            actions << 'remove_border' if may_remove_gauge_change?(entity)
+            actions
+          end
+
+          def setup
+            super
+            @round.removed_gauge = []
+          end
+
+          def round_state
+            super.merge({ removed_gauge: [] })
+          end
+
+          def description
+            tile_lay = get_tile_lay(current_entity)
+            return 'Lay Track' unless tile_lay
+
+            if tile_lay[:lay] && tile_lay[:upgrade]
+              @game.phase.name == 'IV' ? 'Lay/Upgrade Track OR Remove Gauge Change Marker' : 'Lay/Upgrade Track'
+            elsif tile_lay[:lay]
+              'Lay Track'
+            else
+              'Upgrade Track'
+            end
+          end
+
+          # ------ Show a note that gauge changes may be removed ------
+
+          def note_text_block
+            [
+              'May remove a Gauge Change Marker as a track action.',
+              'Click on a connected marker to remove it.',
+            ]
+          end
+
+          def show_note?(entity)
+            may_remove_gauge_change?(entity)
+          end
+
+          # ------ Code for 'remove_gauge_change' Action [Remove Gauge Change Markers in Phase IV] ------
+
+          def may_remove_gauge_change?(_entity)
+            num_track_actions = @round.num_laid_track + @round.num_upgraded_track
+            @game.phase.name == 'IV' && num_track_actions.zero? && any_gauge_changes?
+          end
+
+          def any_gauge_changes?
+            !@game.gauge_change_markers.empty?
+          end
+
+          # A gague change marker is a pair of hexes, which is reachable if both hexes are connected to entity
+          def any_reachable_gauge_changes?(entity)
+            @game.gauge_change_markers.map { |marker| reachable_gauge_change?(entity, marker[0], marker[1]) }.any?
+          end
+
+          def reachable_gauge_change?(entity, hex, neighbor)
+            hex_neighbors(entity, hex) && hex_neighbors(entity, neighbor)
+          end
+
+          # NOTE: Triggered by on_click event in View::Game::Borders
+          def process_remove_border(action)
+            entity = action.entity
+            hex = action.hex
+            tile = hex.tile
+            edge = action.edge
+            neighbor = hex.neighbors[edge]
+            if !@game.loading && !reachable_gauge_change?(entity, hex, neighbor)
+              raise GameError, "#{entity.name} can not reach that marker"
+            end
+
+            tile.borders.reject! { |b| b.edge == edge }
+            neighbor.tile.borders.reject! { |nb| nb.edge == hex.invert(edge) }
```

Comment:
AFAICT, you're not calling the `remove_border_calculate_cost!` method you've added in the RemoveBorders module.

Reply:
I can confirm that the `remove_border_calculate_cost!` method is called.  It is a part of every `lay_tile` action. 
![image](https://github.com/tobymao/18xx/assets/118568589/f7fa0aac-3495-47dd-9d6d-31f3b915bddc)

Reply:
Got it, I was confused with the two different places you had this code for removing borders, but one is for river costs and this one if for the gauge change markers. Right?